### PR TITLE
IPP 12 - Support operationId and PATH_PREFIX

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+  scope "#{ENV['PATH_PREFIX']}/#{ENV['APP_NAME']}" do
     resources :profiles, only: [:index, :show]
     resources :rule_results, only: [:index]
     resources :systems, only: [:index]
     resources :rules, only: [:index, :show]
-    mount Rswag::Api::Engine => '/api-docs'
-    mount Rswag::Ui::Engine => '/api-docs'
-    mount ActionCable.server => '/cable'
+    mount Rswag::Api::Engine => '/'
+    mount Rswag::Ui::Engine => '/'
```
Comment (akofink) [inline config/routes.rb]: Could you give some context here? Our docs are moving to `/api/compliance/` from `/api/compliance/api-docs/`?

File: openshift/templates/compliance-backend.yml [hunk 2]
```diff
@@ -96,15 +98,17 @@ objects:
             value: "${POSTGRESQL_SHARED_BUFFERS}"
           - name: RAILS_ENV
             value: "${RAILS_ENV}"
+          - name: PATH_PREFIX
+            value: "${PATH_PREFIX}"
           livenessProbe:
             httpGet:
-              path: /r/insights/platform/compliance/api-docs/index.html
+              path: /api/compliance/api-docs/index.html
```
Comment (akofink) [inline openshift/templates/compliance-backend.yml]: Couldn't this also be using the env variables?

File: openshift/templates/compliance-backend.yml [hunk 3]
```diff
@@ -96,15 +98,17 @@ objects:
...
               port: 8080
             initialDelaySeconds: 120
             timeoutSeconds: 1
           readinessProbe:
             httpGet:
-              path: /r/insights/platform/compliance/api-docs/index.html
+              path: /api/compliance/api-docs/index.html
```
Comment (akofink) [inline openshift/templates/compliance-backend.yml]: Same here

File: spec/swagger_helper.rb [hunk 4]
```diff
@@ -9,12 +9,12 @@
     'v1/swagger.json' => {
```
Comment (akofink) [inline spec/swagger_helper.rb]: Is it still at `swagger.json`, or now should be `openapi.json`?

File: swagger/v1/openapi.json [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
   "info": {
-    "title": "Insights Compliance API V1",
+    "title": "Cloud Services for RHEL Compliance API V1",
     "version": "v1",
-    "description": "This is the API for Insights Compliance. You can find out more about Red Hat Insights at [https://access.redhat.com/products/red-hat-insights/](https://access.redhat.com/products/red-hat-insights/)"
+    "description": "This is the API for Cloud Services for RHEL Compliance. You can find out more about Red Hat Cloud Services for RHEL at [https://cloud.redhat.com/](https://cloud.redhat.com/)"
   },
   "paths": {
-    "/r/insights/platform/compliance/profiles": {
+    "/api/compliance/profiles": {
```
Comment (akofink) [inline swagger/v1/openapi.json]: Any way to use env variables here?

File: config/routes.rb (hunk 1 above)
Comment (dLobatog) [reply config/routes.rb]: Correct, if you visit the root of the API - it displays the Swagger docs.

File: openshift/templates/compliance-backend.yml (hunk 2 above)
Comment (dLobatog) [reply openshift/templates/compliance-backend.yml]: I'm not sure how, these YAML files are just uploaded to Openshift.

File: spec/swagger_helper.rb (hunk 4 above)
Comment (dLobatog) [reply spec/swagger_helper.rb]: It should be openapi.json as per https://platform-docs.cloud.paas.upshift.redhat.com/rest/requirements.html

File: swagger/v1/openapi.json (hunk 5 above)
Comment (dLobatog) [reply swagger/v1/openapi.json]: This file is auto-generated from the specs, the ENV variables are taken into account directly when running `rake rswag:specs:swaggerize`

File: spec/swagger_helper.rb (hunk 4 above)
Comment (akofink) [reply spec/swagger_helper.rb]: So.. this needs updating, was my point :)

**Pull request merged**